modules/diskq/qdisk.c

```diff
@@ -445,15 +449,75 @@ _load_queue(QDisk *self, GQueue *q, gint64 q_ofs, gint32 q_len, gint32 q_count)
   return TRUE;
 }
 
+#define ignore_queue_error(self, type, ofs) \
+do { \
+  \
+  msg_error("Inconsistent header data in disk-queue file, ignoring " #type, \
+            evt_tag_str("filename", self->filename), \
+            evt_tag_long(#type "_ofs", ofs), \
+            evt_tag_long("write_head", self->hdr->write_head), \
+            evt_tag_long("qdisk_length",  self->hdr->length)); \
+} while (0)
+
+#define _try_to_load_queue(self, queue) \
+( { \
+  gboolean retval = TRUE; \
+  gint64 ofs; \
+  gint count, len; \
+ \
+  count = self->hdr->queue ## _count; \
+  len = self->hdr->queue ## _len; \
+  ofs = self->hdr->queue ## _ofs; \
+ \
+  if (!(ofs > 0 && ofs < self->hdr->write_head)) \
+    { \
+      if (!_load_queue(self, queue, ofs, len, count)) \
+        retval = !self->options->read_only; \
+    } \
+  else \
+    { \
+      ignore_queue_error(self, queue, ofs); \
+    } \
+ \
+  retval; \
+} )
+
+static gboolean
+_load_non_reliable_queues(QDisk *self, GQueue *qout, GQueue *qbacklog, GQueue *qoverflow)
+{
+
+  if (!_try_to_load_queue(self, qout))
+    return FALSE;
+
+  if (!_try_to_load_queue(self, qbacklog))
+    return FALSE;
+
+  if (!_try_to_load_queue(self, qoverflow))
+    return FALSE;
+
+  return TRUE;
+}
+
+static void
+_reset_queue_pointers(QDisk *self)
```

Comment:
We should reset the `..._count` members in the file too, they are already used in load_queue methods,
any later reference in load_state can use the saved values before load_queue.